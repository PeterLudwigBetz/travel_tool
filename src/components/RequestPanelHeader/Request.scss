.request-panel-header {
  box-sizing: border-box;
  background-color: #F8F8F8;
  height: 201.9px;
  width: 96.3%;
  @media (max-width: 479px){
    height: 150.9px;
    width: 94%;
}
  .open-requests {
    display: flex;
    justify-content: space-between;
    align-items: center;
    position: relative;
    top: 50px;
    height: 36px;
  }
}
