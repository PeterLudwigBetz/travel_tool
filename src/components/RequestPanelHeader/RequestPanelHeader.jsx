import React, { PureComponent } from 'react';
import {PropTypes} from 'prop-types';
import Header from '../header/Header';
import ButtonGroup from '../button-group/ButtonGroup';
import HeaderPagination from '../Pagination/HeaderPagination';
import './Request.scss';

class RequestPanelHeader extends PureComponent {
  render() {
    const {toggleNewRequestModal} = this.props;
    return (
      <div className="request-panel-header">
        <Header toggleNewRequestModal={toggleNewRequestModal} />
        <div className="open-requests">
          <ButtonGroup />
          <HeaderPagination />
        </div>
      </div>
    );
  }
}

RequestPanelHeader.propTypes = {
  toggleNewRequestModal: PropTypes.func.isRequired,
};

export default RequestPanelHeader;
