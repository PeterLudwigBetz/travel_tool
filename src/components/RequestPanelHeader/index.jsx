import React, { PureComponent } from 'react';
import Header from '../Header';
import ButtonGroup from '../ButtonGroup';
import HeaderPagination from '../Pagination/HeaderPagination';
import './Request.scss';

class RequestPanelHeader extends PureComponent {
  render() {
    return (
      <div className="request-panel-header">
        <Header />
        <div className="open-requests">
          <ButtonGroup />
          <HeaderPagination />
        </div>
      </div>
    );
  }
}

export default RequestPanelHeader;
