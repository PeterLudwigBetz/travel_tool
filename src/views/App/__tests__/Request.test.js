import React from 'react';
import renderer from 'react-test-renderer';
import RequestPanelHeader from '../../../components/RequestPanelHeader/RequestPanelHeader';

it('should render request header correctly', () => {
  const tree = renderer.create(
    <RequestPanelHeader toggleNewRequestModal={()=>{}} openModal={()=>{}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
