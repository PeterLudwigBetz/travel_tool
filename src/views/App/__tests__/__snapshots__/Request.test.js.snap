// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render request header correctly 1`] = `
<div
  className="request-panel-header"
>
  <div
    className="new-requests"
  >
    <div>
      <span
        className="requests"
      >
        REQUESTS
      </span>
    </div>
    <div>
      <button
        className="btn-new-request"
        onClick={[Function]}
        type="button"
      >
        <span
          className="new-request-button-text"
        >
          New Request
        </span>
      </button>
    </div>
  </div>
  <div
    className="open-requests"
  >
    <div
      className="button-group"
    >
      <button
        className="bg-btn bg-btn--active"
        type="button"
      >
        All
      </button>
      <button
        className="bg-btn bg-btn--with-badge"
        type="button"
      >
        <span
          className="label"
        >
          Open Reqeusts
        </span>
        <span
          className="badge"
        >
          3
        </span>
      </button>
      <button
        className="bg-btn"
        type="button"
      >
        Past Requests
      </button>
    </div>
    <div
      className="cell-item-per-page"
    >
      <div
        className="cell-items-per-page-text"
      >
        <span
          className="items-per-page-text"
        >
          Items per page
        </span>
      </div>
      <div
        className="divider"
      />
      <div
        className="items-per-page-paginator"
      >
        <div
          className="paginator-counter"
        >
          <span
            className="paginator-counter"
          >
            10
          </span>
        </div>
        <div
          className="paginator-counter"
        >
          <span>
            <img
              alt="Andela Logo"
              src="drop-down-icon.svg"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
